### DISCUSSION TITLE
Ignore patterns for fmt and lint

### DISCUSSION THREAD

**Discussion by wperron** [denoland/deno#8784] [Ideas]
Right now we can ignore explicit files and directories with those two commands, but it would be nice to be able to use glob patterns as well.

Bonus: I think we could go a step further and ignore the patterns present in `.gitignore` by default

**Comment by nayeemrmn** [comment]
Duplicate of #6365.

**Comment by caspervonb** [comment]
Related https://github.com/denoland/deno/pull/8664#discussion_r539827312

**Comment by caspervonb** [comment]
> Bonus: I think we could go a step further and ignore the patterns present in .gitignore by default

Probably makes sense, we'd still override that if we provided explicit paths to right? e.g `deno fmt ignored/path.ts` where ignored/path.ts is in .gitignore.

**Reply by wperron** [reply]
that would be consistent with how Git uses `.gitignore` (you can `git add` an ignored file explicitely and it is added to the tree) so my first instinct would be to say yes. I also don't think it would be surprising to users; they clearly added a file path to the command, of course they would expect fmt/lint to apply on that file.

**Comment by caspervonb** [comment]
Ended up not using glob patterns for url matching (regular expressions made more sense, but would have been patterns if that feature wasn't nightly gated).

Personally I prefer my shell to do my expansion for me, that way it works the way I've configured it (e.g expanded globs enabled or not).

Caveat is of course, but what about that one shell in Windows? /shrug

**Comment by wojpawlik** [comment]
I worked around lack of `--ignore-file=.gitignore` (which could contain patterns) by just removing build artifacts before formatting and linting: https://github.com/wojpawlik/deno2node/blob/df9090fd202f3726f2e8687bcd418b20eb08df03/package.json#L16-L17 😛

**Reply by wperron** [reply]
what are we supposed to be looking at here?
